### PR TITLE
[xeb] Wholesale copy of Cirq's grid_parallel_two_qubit_xeb

### PR DESCRIPTION
**Pull request by mpharrigan**
These files are exactly as they appear in cirq.experiments.
I'm commiting them to the feature branch `feature/xeb` to track
the subsequent refactoring into ReCirq idioms. Following
PRs will be against the `feature/xeb` branch.

### CODE REVIEW

**Comment by mpharrigan** [comment]
Actually, there's a minor difference in the tests:

```diff
2a3,4
> import pytest
> 
4,7c6,8
< from cirq.experiments import (collect_grid_parallel_two_qubit_xeb_data,
<                               compute_grid_parallel_two_qubit_xeb_results)
< from cirq.experiments.grid_parallel_two_qubit_xeb import (
<     GridParallelXEBMetadata, LAYER_A, LAYER_B)
---
> from recirq.characterization.xeb.grid_parallel_two_qubit_xeb import GridParallelXEBMetadata, \
>     LAYER_A, LAYER_B, collect_grid_parallel_two_qubit_xeb_data, \
>     compute_grid_parallel_two_qubit_xeb_results
107a109
> @pytest.mark.xfail(reason="`recirq` vs `cirq` in repr.")
```

**Comment by mpharrigan** [comment]
cc @mrwojtek @bfoxen I can't add y'all as reviewers for some reason.

cc @kevinsung if you're interested

File: recirq/characterization/xeb/grid_parallel_two_qubit_xeb_test.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import os
+import numpy as np
+import pytest
+
+import cirq
+from recirq.characterization.xeb.grid_parallel_two_qubit_xeb import GridParallelXEBMetadata, \
+    LAYER_A, LAYER_B, collect_grid_parallel_two_qubit_xeb_data, \
+    compute_grid_parallel_two_qubit_xeb_results
```
Comment (kevinsung) [inline recirq/characterization/xeb/grid_parallel_two_qubit_xeb_test.py:8]: Generally in Cirq we use parentheses for line continuations.

**Review by kevinsung — APPROVED** [review]

File: recirq/characterization/xeb/grid_parallel_two_qubit_xeb.py [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+from cirq import devices, ops, protocols, sim, value
+from cirq.experiments.cross_entropy_benchmarking import (CrossEntropyResult,
+                                                         CrossEntropyResultDict,
+                                                         CrossEntropyPair,
+                                                         SpecklePurityPair)
+from cirq.experiments.fidelity_estimation import (
+    least_squares_xeb_fidelity_from_probabilities)
+from cirq.experiments.purity_estimation import purity_from_probabilities
+from cirq.experiments.random_quantum_circuit_generation import (
+    GridInteractionLayer,
```
Comment (mrwojtek) [inline recirq/characterization/xeb/grid_parallel_two_qubit_xeb.py:44]: Doesn't this also belong to this experiment?

**Review by mrwojtek — APPROVED** [review]
What is the goal of this work?

**Comment by mpharrigan** [comment]
I'm moving the data collection infrastructure for XEB to ReCirq which will leverage the utility functions in `cirq.experiments`. I'll be working on a this feature branch on ReCirq and starting from the existing module `cirq.experiments.grid_parallel_two_qubit_xeb`. This PR copies the code verbatim and I'll refactor in subsequent PRs against that feature branch.

File: recirq/characterization/xeb/grid_parallel_two_qubit_xeb.py (hunk 2 above)
Comment (mpharrigan) [reply recirq/characterization/xeb/grid_parallel_two_qubit_xeb.py:44]: It's always hard to draw the line, but I think there's appetite for keeping the library functionality in Cirq and having the "infrastructure" part that ties it all together in ReCirq.

File: recirq/characterization/xeb/grid_parallel_two_qubit_xeb_test.py (hunk 1 above)
Comment (mpharrigan) [reply recirq/characterization/xeb/grid_parallel_two_qubit_xeb_test.py:10]: changed it

**Pull request merged**